### ISSUE TITLE
Back button support

### ISSUE THREAD

**Issue by murgo** [ubershmekel/redditp#77]
Many times while using redditp.com slideshow I want to go back to previous image and I instinctively hit the back button of my mouse, thus accidentally leaving the site and never finding the place I was at before as going forward just takes me to pic 1.

This could be solved by using the window.history API to push all slideshow images as states, or have the URL represent the state of the slideshow.

I could maybe do this myself and send a PR if this is something you would merge.

**Comment by ubershmekel** [comment]
This is a tricky UX question for me. Because let's say I saw a presentation and went 30 slides in - my back button would be frustrating to use to get out of redditp. I'm not sure what's the right answer here or how to evaluate it. If you have an idea - please do share.

**Comment by murgo** [comment]
I've never been in a position where I wanted the back button to take me away from the page, but then again I use new tabs pretty much.

If this wasn't a single page app it would be expected to have back button to go on previous image.

Still, I understand where you're coming from with the UX concerns and I'm not sure. Maybe check if the back button is handled in other similar web sites.

**Comment by scruffynerf** [comment]
I agree, the expectation is back one image... If someone truly wants a single back to return to previous webpage before redditp, they can either open in New window/tab, or use history... This adds a feature I can see using.

**Comment by murgo** [comment]
Also, if you navigate out from the page via the links and then hit back, it's kinda bad you get redirected to the first image and not where you left off.
